src/Tools/dotnet-format/CodeFormatter.cs

```diff
@@ -74,12 +77,13 @@ public static async Task<int> FormatWorkspaceAsync(ILogger logger, string soluti
 
         private static async Task<int> FormatFilesInSolutionAsync(ILogger logger, Solution solution, string projectPath, ICodingConventionsManager codingConventionsManager, CancellationToken cancellationToken)
         {
-            var formattedSolution = solution;
+            var workspace = solution.Workspace;
+            var projectIds = workspace.CurrentSolution.ProjectIds.ToImmutableArray();
```

Comment:
neither the local nor the immutablearray seem necessary.

Reply:
ok.  this api just seems weird.  we're given an Solution, but we then use that to get teh Workspace... which we then use to get teh CurrentSolution..

Reply:
`ProjectIds` returns an `IReadOnlyCollection<T>`, which generally isn't safe for iteration during mutating operations. It might be safe in this circumstance, but that's still a weaker guarantee than the `ImmutableArray<T>` provides.

I'll update the implementation to pass `Workspace` instead of `Solution` to this method.

Reply:
➡️ Fixed by amending the last commit.

Reply:
> ProjectIds returns an IReadOnlyCollection<T>, which generally isn't safe for iteration during mutating operations

It shoudl def be safe though as per the roslyn Worspace invariants.  The ProjectIds of a Solution cannot change out from underneath you.  That would be a gross violation of the contract of Solution :)